x-pack/plugins/security_solution/cypress/integration/detection_rules/bulk_edit_rules.spec.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ELASTIC_RULES_BTN,
+  CUSTOM_RULES_BTN,
+  MODAL_CONFIRMATION_BTN,
+  SELECT_ALL_RULES_ON_PAGE_CHECKBOX,
+  LOAD_PREBUILT_RULES_ON_PAGE_HEADER_BTN,
+  RULES_TAGS_FILTER_BTN,
+} from '../../screens/alerts_detection_rules';
+
+import {
+  RULES_BULK_EDIT_OVERWRITE_INDEX_PATTERNS_CHECKBOX,
+  RULES_BULK_EDIT_OVERWRITE_TAGS_CHECKBOX,
+  RULES_BULK_EDIT_INDEX_PATTERNS_WARNING,
+  RULES_BULK_EDIT_TAGS_WARNING,
+} from '../../screens/rules_bulk_edit';
+
+import {
+  changeRowsPerPageTo,
+  waitForRulesTableToBeLoaded,
+  selectAllRules,
+  goToTheRuleDetailsOf,
+  waitForRulesTableToBeRefreshed,
+  selectNumberOfRules,
+  testAllTagsBadges,
+} from '../../tasks/alerts_detection_rules';
+
+import {
+  openBulkEditAddIndexPatternsForm,
+  openBulkEditDeleteIndexPatternsForm,
+  typeIndexPatterns,
+  waitForBulkEditActionToFinish,
+  confirmBulkEditForm,
+  clickAddIndexPatternsMenuItem,
+  waitForElasticRulesBulkEditModal,
+  waitForMixedRulesBulkEditModal,
+  openBulkEditAddTagsForm,
+  openBulkEditDeleteTagsForm,
+  typeTags,
+} from '../../tasks/rules_bulk_edit';
+
+import { hasIndexPatterns } from '../../tasks/rule_details';
+import { loginAndWaitForPageWithoutDateRange } from '../../tasks/login';
+
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+import { createCustomRule } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import {
+  getExistingRule,
+  getNewOverrideRule,
+  getNewRule,
+  getNewThresholdRule,
+  totalNumberOfPrebuiltRules,
+} from '../../objects/rule';
+
+const RULE_NAME = 'Custom rule for bulk actions';
+
+const CUSTOM_INDEX_PATTERN_1 = 'custom-cypress-test-*';
+const DEFAULT_INDEX_PATTERNS = ['index-1-*', 'index-2-*'];
+const TAGS = ['cypress-tag-1', 'cypress-tag-2'];
+const OVERWRITE_INDEX_PATTERNS = ['overwrite-index-1-*', 'overwrite-index-2-*'];
+
+const customRule = {
+  ...getNewRule(),
+  index: DEFAULT_INDEX_PATTERNS,
+  name: RULE_NAME,
+};
+
+describe('Detection rules, bulk edit', () => {
+  beforeEach(() => {
+    cleanKibana();
+
+    createCustomRule(customRule, '1');
+    createCustomRule(getExistingRule(), '2');
+    createCustomRule(getNewOverrideRule(), '3');
+    createCustomRule(getNewThresholdRule(), '4');
+    createCustomRule({ ...getNewRule(), name: 'rule # 5' }, '5');
+    createCustomRule({ ...getNewRule(), name: 'rule # 6' }, '6');
+
+    loginAndWaitForPageWithoutDateRange(SECURITY_DETECTIONS_RULES_URL);
+    waitForRulesTableToBeLoaded();
+  });
+
+  it('should show modal windows when Elastic rules selected and edit only custom rules', () => {
+    cy.get(LOAD_PREBUILT_RULES_ON_PAGE_HEADER_BTN)
```

Comment:
It would e nice if you can wrap all these methods that are not assertions into tasks methods that reveal the intention of the user in order to follow our good practices when writing Cypress tests.

Reply:
thanks, I will look into it in the next iterations of test for bulk edit. We are looking to release new bulk edit updates in next releases